src/Kmd.Logic.CitizenDocuments.Client/CitizenDocumentsClient.cs

```diff
@@ -89,6 +94,144 @@ public async Task<CitizenDocumentUploadResponse> UploadAttachmentWithHttpMessage
+        /// <exception cref="LogicTokenProviderException">Unable to issue an authorization token.</exception>
+        /// <exception cref="CitizenDocumentsException">Invalid Citizen document configuration details.</exception>
+        public async Task<CitizenDocumentUploadResponse> UploadFileAttachmentDirectlyToStorageAsync(Stream document, CitizenDocumentUploadRequestModel citizenDocumentUploadRequestModel)
+        {
+            if (document == null)
+            {
+                throw new ArgumentNullException(nameof(document));
+            }
+
+            if (citizenDocumentUploadRequestModel == null)
+            {
+                throw new ArgumentNullException(nameof(citizenDocumentUploadRequestModel));
+            }
+
+            if (citizenDocumentUploadRequestModel.CitizenDocumentConfigId == null)
+            {
+                throw new Exception("Citizen Document Configuration is null");
+            }
+
+            if (citizenDocumentUploadRequestModel.Cpr == null)
+            {
+                throw new Exception("Cpr is null");
+            }
+
+            if (citizenDocumentUploadRequestModel.DocumentName == null)
+            {
+                throw new Exception("DocumentName is null");
+            }
+
+            if (citizenDocumentUploadRequestModel.RetentionPeriodInDays == null)
+            {
+                throw new Exception("RetentionPeriodInDays is null");
+            }
+
+            if (citizenDocumentUploadRequestModel.DocumentType == null)
+            {
+                throw new Exception("DocumentType is null");
+            }
+
+            var client = this.CreateClient();
+            var documentId = Guid.NewGuid();
+            var storageDocName = citizenDocumentUploadRequestModel.DocumentName.Trim().Replace(".", string.Empty) + "_" + documentId + ".pdf";
+            var responseSasUri = await client.StorageAccessWithHttpMessagesAsync(
+                                   subscriptionId: new Guid(this.options.SubscriptionId),
+                                   documentName: storageDocName).ConfigureAwait(false);
+
+            if (responseSasUri.Response.StatusCode != System.Net.HttpStatusCode.OK)
+            {
+                throw new Exception("You don't have permission to upload");
+            }
+
+            var sasTokenUri = new Uri(responseSasUri.Body);
+
+            var containerAddress = new Uri($"{sasTokenUri.Scheme}://{sasTokenUri.Host}/{sasTokenUri.AbsolutePath.Split('/')[1]}");
+
+            CloudBlobContainer container = new CloudBlobContainer(
+                containerAddress,
+                new StorageCredentials(sasTokenUri.Query));
+
+            var uploadResponse = await UploadDocumentToAzureStorage(document, storageDocName, container, 5000000).ConfigureAwait(false);
+
+            if (uploadResponse.StatusCode != System.Net.HttpStatusCode.OK)
+            {
+                throw new Exception("Upload Failed");
```

Comment:
`Exception` is base type and it should never be throws. Please use more specific type for exception.

Reply:
Changed exception